services/ux-backend/handlers/onboarding_tokens.go

```diff
@@ -0,0 +1,120 @@
+package handler
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/red-hat-storage/ocs-operator/v4/services/types"
+	"k8s.io/klog/v2"
+)
+
+const onboardingPrivateKeyFilePath = "/etc/private-key/key"
+
+func OnboardingTokensHandler(w http.ResponseWriter, r *http.Request, tokenLifetimeInHours int) {
+
+	var err error
+	switch r.Method {
+	case "POST":
+
+		tokenExpirationDate := time.Now().
+			Add(time.Duration(tokenLifetimeInHours) * time.Hour).
+			Unix()
+
+		onboardingToken, err := generateOnboardingToken(tokenExpirationDate)
+		if err != nil {
+			klog.Error(err, "failed to get onboardig token")
+			w.WriteHeader(http.StatusInternalServerError)
+			w.Header().Set("Content-Type", "text/text")
+			_, err = w.Write([]byte("Failed to generate token"))
+
+			if err != nil {
+				klog.Error(err, "failed write data to response writer")
+			}
+			return
+		}
+
+		klog.Info("onboarding token generated successfully")
+		w.WriteHeader(http.StatusOK)
+		w.Header().Set("Content-Type", "text/text")
+
+		_, err = w.Write([]byte(onboardingToken))
+		if err != nil {
+			klog.Error(err, "failed write data to response writer")
+			return
+		}
+
+	default:
+		klog.Info("Only POST method should be used to send data to this endpoint /onboarding-tokens")
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		w.Header().Set("Content-Type", "text/text")
+		_, err = w.Write([]byte(fmt.Sprintf("Unsupported method : %s", r.Method)))
+		if err != nil {
+			klog.Error(err, "failed write data to response writer")
```

Comment:
By logging the error first you relegate the issue to the last thing in the message. Preferably use:
```suggestion
			klog.Errorf("failed write data to response writer: %v", err)
```